tools/link.py

```diff
@@ -166,18 +166,17 @@ def setup_environment_settings():
   # Note: we need to actually modify ENVIRONMENTS variable here before the parsing,
   # because some JS code reads it back so modifying parsed info alone is not sufficient.
   if settings.SHARED_MEMORY and settings.ENVIRONMENT:
-    settings.ENVIRONMENT += ',worker'
+    settings.ENVIRONMENT.append('worker')
 
   # Environment setting based on user input
-  environments = settings.ENVIRONMENT.split(',')
-  if any(x for x in environments if x not in VALID_ENVIRONMENTS):
+  if any(x for x in settings.ENVIRONMENT if x not in VALID_ENVIRONMENTS):
     exit_with_error(f'Invalid environment specified in "ENVIRONMENT": {settings.ENVIRONMENT}. Should be one of: {",".join(VALID_ENVIRONMENTS)}')
 
-  settings.ENVIRONMENT_MAY_BE_WEB = not settings.ENVIRONMENT or 'web' in environments
-  settings.ENVIRONMENT_MAY_BE_WEBVIEW = not settings.ENVIRONMENT or 'webview' in environments
-  settings.ENVIRONMENT_MAY_BE_NODE = not settings.ENVIRONMENT or 'node' in environments
-  settings.ENVIRONMENT_MAY_BE_SHELL = not settings.ENVIRONMENT or 'shell' in environments
-  settings.ENVIRONMENT_MAY_BE_WORKER = not settings.ENVIRONMENT or 'worker' in environments
+  settings.ENVIRONMENT_MAY_BE_WEB = not settings.ENVIRONMENT or 'web' in settings.ENVIRONMENT
```

Comment:
Btw, I didn't want to touch those, but... do we actually need all those `not settings.ENVIRONMENT`? I assume it existed to handle an empty default, but our default list is already non-empty...

Reply:
> Setting this to the empty list explictly means "all environments".

Ah, as in "even more than the default"? Feels a bit obscure, but yeah, then it would be a breaking change, nvm.